### PR TITLE
Added document for install federation v2 on IBM Cloud Private.

### PR DESCRIPTION
**Pull request by gyliu513**

### CODE REVIEW

File: docs/environments/icp.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# IBM Cloud Private Deployment Guide
+
+Federation v2 can be deployed to and manage [IBM Cloud Private](https://www.ibm.com/cloud/private) clusters.
+As Federation v2 requests Kubernetes v1.11 or greater, so please make sure deploy IBM Cloud Private 3.1.1
```
Comment (marun) [inline docs/environments/icp.md]: `s/requests/requires/`
`s/so please make sure deploy/please make sure to deploy/`

File: docs/environments/icp.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+or higher version.
```
Comment (marun) [inline docs/environments/icp.md]: `s/version//`

File: docs/environments/icp.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The following example deploys two IBM Cloud Private 3.1.1 clusters named as `cluster1` and `cluster2`.
```
Comment (marun) [inline docs/environments/icp.md]: `s/as//`

File: docs/environments/icp.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Install IBM Cloud Private
+
+Please follow the [Guidance in IBM Cloud Private 3.1.1 Knowledge Center](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install.html)
```
Comment (marun) [inline docs/environments/icp.md]: `s/Guidance/guide/`

File: docs/environments/icp.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+to install.
+
+**NOTE:** As we need to install two clusters named as `cluster1` and `cluster2`, so after `cluster/config.yaml`
```
Comment (marun) [inline docs/environments/icp.md]: `s/as//`

File: docs/environments/icp.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+generated, update the cluster name to `cluster1` and `cluster2` for the two clusters before install.
```
Comment (marun) [inline docs/environments/icp.md]: `s/generated/is generated/`
`s/update the cluster name.*/update the names of the 2 clusters to 'cluster1' and 'cluster2' before installing federation`

File: docs/environments/icp.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+For the second cluster, set the following value in `cluster/config.yaml` as follows:
+
+```yaml
+cluster_name: cluster2
+```
+
+## Post Install Configuration
+
+As IBM Cloud Private was [enforcing container image security](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/manage_images/image_security.html)
```
Comment (marun) [inline docs/environments/icp.md]: `s/was/is/`

File: docs/environments/icp.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+policy by default, and the default image security policy do not allow pull Federation v2 image
+from `quay.io/kubernetes-multicluster/federation-v2:*`, so we need to update the image security
```
Comment (marun) [inline docs/environments/icp.md]: `s/so//`

File: docs/environments/icp.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+For the second cluster, set the following value in `cluster/config.yaml` as follows:
+
+```yaml
+cluster_name: cluster2
+```
+
+## Post Install Configuration
+
+As IBM Cloud Private was [enforcing container image security](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/manage_images/image_security.html)
+policy by default, and the default image security policy do not allow pull Federation v2 image
```
Comment (marun) [inline docs/environments/icp.md]: `s/do/does/`
`s/pull/pulling the/`

Comment (gyliu513) [reply docs/environments/icp.md]: Done

File: docs/environments/icp.md (hunk 7 above)
Comment (gyliu513) [reply docs/environments/icp.md]: Done

File: docs/environments/icp.md (hunk 8 above)
Comment (gyliu513) [reply docs/environments/icp.md]: Done

File: docs/environments/icp.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+or higher.
+
+The following example deploys two IBM Cloud Private 3.1.1 clusters named `cluster1` and `cluster2`.
+
+## Install IBM Cloud Private
+
+Please follow the [guide in IBM Cloud Private 3.1.1 Knowledge Center](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install.html)
+to install.
+
+**NOTE:** As we need to install two clusters named `cluster1` and `cluster2`, so after `cluster/config.yaml`
```
Comment (marun) [inline docs/environments/icp.md]: s/As we/We/

File: docs/environments/icp.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+is generated, update the names of the 2 clusters to 'cluster1' and 'cluster2' before installing federation.
```
Comment (marun) [inline docs/environments/icp.md]: (No action required) Should this be \`cluster1\` and \`cluster2\` for consistency?

**Comment by gyliu513** [comment]
/cc @pmorie @irfanurrehman 

Can you help lgtm if no further comments as @marun is in vacation now?

**Comment by marun** [comment]
/lgtm

Thank you for your patience!

**Pull request merged**